test/fmf/tests/utils.sh

```diff
@@ -2,10 +2,81 @@
 
 set -euo pipefail
 
-# We don't need to generate the certificates as they are generated
-# by the systemd services if they don't exist
+# USE_TLS should be set before sourcing this file
+: "${USE_TLS:=false}"
+
+# Set up curl insecure flag for TLS mode
+if [ "$USE_TLS" = true ]; then
+  export CURL_INSECURE_FLAG="--insecure"
+else
+  export CURL_INSECURE_FLAG=""
+fi
+
+# Generate TLS certificates for HTTPS transport (TLS mode only)
+# These are separate from the FDO protocol certificates (manufacturer/owner keys)
+generate_tls_cert() {
+  local service_name=$1
+  local conf_dir=$2
+  local subj=$3
+  local user="go-fdo-server-${service_name}"
+  local group="go-fdo-server"
+  
+  local key="${conf_dir}/${service_name}-tls.key"
+  local crt="${conf_dir}/${service_name}-tls.crt"
+  
+  if [[ ! -f "${key}" && ! -f "${crt}" ]]; then
+    [ -d "${conf_dir}" ] || sudo mkdir -p "${conf_dir}"
+    
+    # Generate private key in PEM format (required by Go's TLS)
+    openssl ecparam -name prime256v1 -genkey -out "${key}"
+    
+    # Generate self-signed certificate
+    openssl req -new -x509 -key "${key}" -subj "${subj}" -days 365 -out "${crt}"
+    
+    # Set ownership to the service user and go-fdo-server group
+    sudo chown "${user}:${group}" "${key}" "${crt}"
+    sudo chmod 640 "${key}"
+    sudo chmod 644 "${crt}"
+  fi
+}
```

Comment:
Not needed. this is basically `generate_cert` from `scripts/cert-utils.sh`. I just realized we do not need the pub key file anymore so feel free to modify the function to remove it.

Reply:
it's needed cause generate_cert generates der keys, and we need pem - unless you want to overload this function with a type which I can do but I don't like, let's see how it is